main/test/flix/fuzzers/FuzzSwapLines.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 Magnus Madsen, Herluf Baggesen
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package flix.fuzzers
+
+import ca.uwaterloo.flix.TestUtils
+import ca.uwaterloo.flix.api.Flix
+import org.scalatest.funsuite.AnyFunSuite
+
+import java.nio.file.{Files, Paths}
+import scala.jdk.CollectionConverters._
+
+class FuzzSwapLines extends AnyFunSuite with TestUtils {
+
+  // Swap lines takes a _long_ time to run.
+  // In a file with just 100 lines, there is (100 * 99) / 2 unique non-equal swaps.
+  // That's 4950 compiles if we try them all.
+  // Assuming each take 1sec to run that ends up at 1.3 hours.
+  // Instead we select numSwapLines and try to swap those with each-other.
+  // For instance numSwapLines = 10 gives a total of 330 swaps per file.
+  private val numSwapLines = 10
+
+  test("simple-card-game") {
+    val lines = Files.lines(Paths.get("examples/simple-card-game.flix"))
+    compileWithSwappedLines("simple-card-game", lines)
+  }
+
+  test("using-channels-and-select") {
+    val lines = Files.lines(Paths.get("examples/using-channels-and-select.flix"))
+    compileWithSwappedLines("using-channels-and-select", lines)
+  }
+
+  test("the-ast-typing-problem-with-polymorphic-records") {
+    val lines = Files.lines(Paths.get("examples/the-ast-typing-problem-with-polymorphic-records.flix"))
```

Comment:
Could we store the path once and then use path.getFileName().toString?

Reply:
Like so?
```scala
test("simple-card-game") {
    val filepath = Paths.get("examples/simple-card-game.flix")
    val lines = Files.lines(filepath)
    compileWithDuplicateLine(filepath.getFileName.toString, lines)
  }
```

Reply:
Yeah